hittingEndPoint.vb

```diff
@@ -0,0 +1,176 @@
+            oauth2.SleepMs(100)
+            numMsWaited = numMsWaited + 100
+        End While
+
+        If (oauth2.AuthFlowState < 3) Then
+            oauth2.Cancel()
+            MessageBox.Show("No response from the browser!")
+            Return ""
+        End If
+        If (oauth2.AuthFlowState = 5) Then
+            MessageBox.Show("OAuth2 failed to complete.")
+            MessageBox.Show(oauth2.FailureInfo)
+            Return ""
+        End If
+        If (oauth2.AuthFlowState = 4) Then
+            MessageBox.Show("OAuth2 authorization was denied.")
+            MessageBox.Show(oauth2.AccessTokenResponse)
+            Return ""
+        End If
+        If (oauth2.AuthFlowState <> 3) Then
+            MessageBox.Show("Unexpected AuthFlowState:" & oauth2.AuthFlowState)
+            Return ""
+        End If
+        Return oauth2.AccessToken
+    End Function
+
+    Public Function callSheetsAPI(ByRef fileId As String, ByRef userClientId As String , ByRef userClientSecretId As String) As List(Of String)
+        Dim lines As List(Of String) = Nothing
+        'Calls for Authorization Token.
+        Dim _bearerToken As String = getAuthorizationToken(userClientId,userClientSecretId)
+        If _bearerToken <> "" Then
+            Dim myUri As New Uri("https://docs.google.com/spreadsheets/vbaprocessfile?fid=" + fileId)
+            Dim myWebRequest = System.Net.HttpWebRequest.Create(myUri)
+            Dim myHttpWebRequest = CType(myWebRequest, System.Net.HttpWebRequest)
+            myHttpWebRequest.Method = "GET"
+            myHttpWebRequest.PreAuthenticate = True
+            myHttpWebRequest.Headers.Add("Authorization", "Bearer " & _bearerToken)
+            Dim myWebResponse As HttpWebResponse = myWebRequest.GetResponse()
+            'Collectting the response after request.
+            Dim responseStream As Stream = myWebResponse.GetResponseStream()
+            If responseStream Is Nothing Then
+                MessageBox.Show("Api didn't respond.")
+                Return lines
+            End If
+            'Creating a file to store data from the response stream.
+            Dim Json As New FileStream("D:\Json.txt", FileMode.Create)
+            Dim read As Byte() = New Byte(255) {}
+            Dim count As Integer = responseStream.Read(read, 0, read.Length)
+            'Writing the response data in the file created above.
+            While count > 0
+                Json.Write(read, 0, count)
+                count = responseStream.Read(read, 0, read.Length)
+            End While
+            'Closing the variables.
+            Json.Close()
+            responseStream.Close()
+            myWebResponse.Close()
+            'Calling the function "parseTheFile()" to get the information needed from the downloaded file.
+            lines = parseTheFile()
+        End If
+        Return lines
+    End Function
+
+    Private Function parseTheFile() As List(Of String)

```

Comment:
Please make this function modular.

Reply:
Divided it in two part. 
1. Read the file.
2. Parse it.

Reply:
Good. Thanks.